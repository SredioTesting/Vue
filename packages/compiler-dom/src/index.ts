import { baseCompile, CompilerOptions, CodegenResult } from '@vue/compiler-core'
import { parserOptionsMinimal } from './parserOptionsMinimal'
import { parserOptionsStandard } from './parserOptionsStandard'
import { transformStyle } from './transforms/transformStyle'
import { transformText } from './transforms/vText'

export function compile(
  template: string,
  options: CompilerOptions = {}
): CodegenResult {
  return baseCompile(template, {
    ...options,
    ...(__BROWSER__ ? parserOptionsMinimal : parserOptionsStandard),
    nodeTransforms: [
      transformStyle,
      transformText,
      ...(options.nodeTransforms || [])
    ],
    directiveTransforms: {
      // TODO include DOM-specific directiveTransforms
      ...(options.directiveTransforms || {})
    }
  })
}

export * from '@vue/compiler-core'
