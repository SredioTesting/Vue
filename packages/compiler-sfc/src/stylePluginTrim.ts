import { Rule, AtRule } from 'postcss'

const handleTrim = ({ raws }: Rule | AtRule) => {
  if (raws.before) raws.before = '\n'
  if (raws.after) raws.after = '\n'
}

const trimPlugin = () => ({
  postcssPlugin: 'trim',
  Rule: handleTrim,
  AtRule: handleTrim
})
trimPlugin.postcss = true

export default trimPlugin
