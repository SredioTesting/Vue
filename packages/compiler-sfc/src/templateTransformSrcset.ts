import path from 'path'
import {
  ConstantTypes,
  createCompoundExpression,
  createSimpleExpression,
  NodeTransform,
  NodeTypes,
  SimpleExpressionNode
} from '@vue/compiler-core'
import {
  isRelativeUrl,
  parseUrl,
  isExternalUrl,
  isDataUrl
} from './templateUtils'
import {
  AssetURLOptions,
  defaultAssetUrlOptions
} from './templateTransformAssetUrl'

const srcsetTags = ['img', 'source']

interface ImageCandidate {
  url: string
  descriptor: string
}

// http://w3c.github.io/html/semantics-embedded-content.html#ref-for-image-candidate-string-5
const escapedSpaceCharacters = /( |\\t|\\n|\\f|\\r)+/g

export const createSrcsetTransformWithOptions = (
  options: Required<AssetURLOptions>
): NodeTransform => {
  return (node, context) =>
    (transformSrcset as Function)(node, context, options)
}

export const transformSrcset: NodeTransform = (
  node,
  context,
  options: Required<AssetURLOptions> = defaultAssetUrlOptions
) => {
  if (node.type === NodeTypes.ELEMENT) {
    if (srcsetTags.includes(node.tag) && node.props.length) {
      node.props.forEach((attr, index) => {
        if (attr.name === 'srcset' && attr.type === NodeTypes.ATTRIBUTE) {
          if (!attr.value) return
          const value = attr.value.content
          if (!value) return
          const imageCandidates: ImageCandidate[] = value.split(',').map(s => {
            // The attribute value arrives here with all whitespace, except
            // normal spaces, represented by escape sequences
            const [url, descriptor] = s
              .replace(escapedSpaceCharacters, ' ')
              .trim()
              .split(' ', 2)
            return { url, descriptor }
          })

          // data urls contains comma after the ecoding so we need to re-merge
          // them
          for (let i = 0; i < imageCandidates.length; i++) {
            const { url } = imageCandidates[i]
            if (isDataUrl(url)) {
              imageCandidates[i + 1].url =
                url + ',' + imageCandidates[i + 1].url
              imageCandidates.splice(i, 1)
            }
          }

          const shouldProcessUrl = (url: string) => {
            return (
              !isExternalUrl(url) &&
              !isDataUrl(url) &&
              (options.includeAbsolute || isRelativeUrl(url))
            )
          }

          const hasQualifiedUrl = imageCandidates.some(({ url }) => shouldProcessUrl(url))
          // When srcset does not contain any qualified URLs, skip transforming
          if (!hasQualifiedUrl) {
            return
          }

          if (options.base) {
            const base = options.base
            const set: string[] = []
            let needHoistPath = false
            imageCandidates.forEach((imageCandidate) => {
              let { descriptor, url } = imageCandidate
              descriptor = descriptor ? ` ${descriptor}` : ``
              if (url[0] === '.') {
                imageCandidate.url = (path.posix || path).join(base, url)
                set.push(imageCandidate.url + descriptor)
              } else if (shouldProcessUrl(url)) {
                needHoistPath = true
              } else {
                set.push(url + descriptor)
              }
            })

            if (!needHoistPath) {
              attr.value.content = set.join(', ')
              return
            }
          }

          const compoundExpression = createCompoundExpression([], attr.loc)
          imageCandidates.forEach(({ url, descriptor }, index) => {
            if (shouldProcessUrl(url)) {
              const { path } = parseUrl(url)
              let exp: SimpleExpressionNode
              if (path) {
                const existingImportsIndex = context.imports.findIndex(
                  i => i.path === path
                )
                if (existingImportsIndex > -1) {
                  exp = createSimpleExpression(
                    `_imports_${existingImportsIndex}`,
                    false,
                    attr.loc,
                    ConstantTypes.CAN_HOIST
                  )
                } else {
                  exp = createSimpleExpression(
                    `_imports_${context.imports.length}`,
                    false,
                    attr.loc,
                    ConstantTypes.CAN_HOIST
                  )
                  context.imports.push({ exp, path })
                }
                compoundExpression.children.push(exp)
              }
            } else {
              const exp = createSimpleExpression(
                `"${url}"`,
                false,
                attr.loc,
                ConstantTypes.CAN_HOIST
              )
              compoundExpression.children.push(exp)
            }
            const isNotLast = imageCandidates.length - 1 > index
            if (descriptor && isNotLast) {
              compoundExpression.children.push(` + ' ${descriptor}, ' + `)
            } else if (descriptor) {
              compoundExpression.children.push(` + ' ${descriptor}'`)
            } else if (isNotLast) {
              compoundExpression.children.push(` + ', ' + `)
            }
          })

          const hoisted = context.hoist(compoundExpression)
          hoisted.constType = ConstantTypes.CAN_HOIST

          node.props[index] = {
            type: NodeTypes.DIRECTIVE,
            name: 'bind',
            arg: createSimpleExpression('srcset', true, attr.loc),
            exp: hoisted,
            modifiers: [],
            loc: attr.loc
          }
        }
      })
    }
  }
}
