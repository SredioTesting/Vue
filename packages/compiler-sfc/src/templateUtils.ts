import { UrlWithStringQuery, parse as uriParse } from 'url'
import { isString } from '@vue/shared'

export function isRelativeUrl(url: string): boolean {
  return url.startsWith('.') || url.startsWith('~') || url.startsWith('@')
}

// We need an extra transform context API for injecting arbitrary import
// statements.
export function parseUrl(url: string): UrlWithStringQuery {
  if (url.startsWith('~')) {
    const secondChar = url.charAt(1)
    url = url.slice(secondChar === '/' ? 2 : 1)
  }
  return parseUriParts(url)
}

/**
 * vuejs/component-compiler-utils#22 Support uri fragment in transformed require
 * @param urlString an url as a string
 */
function parseUriParts(urlString: string): UrlWithStringQuery {
  // A TypeError is thrown if urlString is not a string
  // @see https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost
  return uriParse(isString(urlString) ? urlString : '')
}
