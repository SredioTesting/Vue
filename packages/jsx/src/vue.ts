import type { VNodeRef } from '@vue/runtime-core'

export type ReservedProps = {
  key?: string | number | symbol
  ref?: VNodeRef
  ref_for?: boolean
  ref_key?: string
}
