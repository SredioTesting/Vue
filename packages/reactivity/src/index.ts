export {
  ref,
  shallowRef,
  isRef,
  toRef,
  toValue,
  toRefs,
  unref,
  proxyRefs,
  customRef,
  triggerRef,
  type Ref,
  type MaybeRef,
  type MaybeRefOrGetter,
  type ToRef,
  type ToRefs,
  type UnwrapRef,
  type ShallowRef,
  type ShallowUnwrapRef,
  type RefUnwrapBailTypes,
  type CustomRefFactory,
} from './ref'
export {
  reactive,
  readonly,
  isReactive,
  isReadonly,
  isShallow,
  isProxy,
  shallowReactive,
  shallowReadonly,
  markRaw,
  toRaw,
  type Raw,
  type DeepReadonly,
  type ShallowReactive,
  type UnwrapNestedRefs,
} from './reactive'
export {
  computed,
  type ComputedRef,
  type WritableComputedRef,
  type WritableComputedOptions,
  type ComputedGetter,
  type ComputedSetter,
  type ComputedRefImpl,
} from './computed'
export {
  effect,
  stop,
  enableTracking,
  pauseTracking,
  resetTracking,
  ReactiveEffect,
  EffectFlags,
  type ReactiveEffectRunner,
  type ReactiveEffectOptions,
  type EffectScheduler,
  type DebuggerOptions,
  type DebuggerEvent,
  type DebuggerEventExtraInfo,
} from './effect'
export { trigger, track, ITERATE_KEY } from './dep'
export {
  effectScope,
  EffectScope,
  getCurrentScope,
  onScopeDispose,
} from './effectScope'
export { TrackOpTypes, TriggerOpTypes, ReactiveFlags } from './constants'
