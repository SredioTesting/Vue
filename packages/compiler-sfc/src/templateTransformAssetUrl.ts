import {
  AttributeNode,
  createSimpleExpression,
  ExpressionNode,
  NodeTransform,
  NodeTypes,
  SourceLocation,
  TransformContext
} from '@vue/compiler-core'
import { parseUrl } from './templateUtils'

export interface AssetURLOptions {
  [name: string]: string[]
}

const assetURLOptions: AssetURLOptions = {
  video: ['src', 'poster'],
  source: ['src'],
  img: ['src'],
  image: ['xlink:href', 'href'],
  use: ['xlink:href', 'href']
}

export const transformAssetUrl: NodeTransform = (node, context) => {
  if (node.type === NodeTypes.ELEMENT) {
    for (const tag in assetURLOptions) {
      if ((tag === '*' || node.tag === tag) && node.props.length) {
        const attributes = assetURLOptions[tag]
        attributes.forEach(item => {
          node.props.forEach((attr: AttributeNode, index) => {
            if (attr.type !== NodeTypes.ATTRIBUTE) return
            if (attr.name !== item) return
            if (!attr.value) return
            const url = parseUrl(attr.value.content)
            const exp = getImportsExpressionExp(
              url.path,
              url.hash,
              attr.loc,
              context
            )
            node.props[index] = {
              type: NodeTypes.DIRECTIVE,
              name: 'bind',
              arg: createSimpleExpression(item, true, attr.loc),
              exp,
              modifiers: [],
              loc: attr.loc
            }
          })
        })
      }
    }
  }
}

function getImportsExpressionExp(
  path: string | undefined,
  hash: string | undefined,
  loc: SourceLocation,
  context: TransformContext
): ExpressionNode {
  if (path) {
    const index = getImportsPathIndex(path, context)
    if (index > -1) {
      return Array.from(context.imports).map(o => o.exp)[
        index
      ] as ExpressionNode
    }
    const name = `_imports_${context.imports.size}`
    const exp = createSimpleExpression(name, false, loc, true)
    context.imports.add({ exp, path })
    if (hash && path) {
      return context.hoist(
        createSimpleExpression(`${name} + '${hash}'`, false, loc, true)
      )
    } else {
      return exp
    }
  } else {
    return createSimpleExpression(`''`, false, loc, true)
  }
}

function getImportsPathIndex(path: string, context: TransformContext): number {
  return Array.from(context.imports).findIndex(o => o.path === path)
}
