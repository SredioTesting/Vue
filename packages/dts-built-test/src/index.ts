import { type PropType, defineComponent } from 'vue'

const _CustomPropsNotErased = defineComponent({
  props: {},
  setup() {},
})

export const RegularComponent = defineComponent({
  props: {},
  setup() {
    return () => {}
  },
})

type MyInterface =
  | {
      a: string
    }
  | { b: string }
export const RegularComponentProps = defineComponent({
  props: {
    a: Object as () => MyInterface,
    b: Object as PropType<MyInterface>,
  },
  setup(props) {
    return () => {}
  },
})
// #8376
export const CustomPropsNotErased =
  _CustomPropsNotErased as typeof _CustomPropsNotErased & {
    foo: string
  }
