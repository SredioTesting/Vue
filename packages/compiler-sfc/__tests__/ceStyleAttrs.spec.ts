import { parse } from '../src'
import { assertCode, compileSFCScript } from './utils'

describe('CE style attrs injection', () => {
  test('generating correct code for nested paths', () => {
    const { content } = compileSFCScript(
      `<script>const a = 1</script>\n` +
        `<style :id="{ id: msg3, 'other-attr': msg }"
id="{ id: msg2, 'other-attr': msg }"
.id="{ id: msg, 'other-attr': msg }"
v-bind:src="msg"
v-bind:[msg]="msg"
:[msg]="msg"
:xlink:special="msg">div{
          color: red;
        }</style>`,
    )

    expect(content).toMatch(`_useCEStyleAttrs(_ctx => ([
  {
   "id": { id: _ctx.msg3, 'other-attr': _ctx.msg },
   "src": _ctx.msg,
   [_ctx.msg]: _ctx.msg,
   "xlink:special": _ctx.msg,
 },
]))}`)
    assertCode(content)
  })

  test('w/ normal <script> binding analysis', () => {
    const { content } = compileSFCScript(
      `<script>
      export default {
        setup() {
          return {
            msg: ref('100px')
          }
        }
      }
      </script>\n` +
        `<style :id="{ id: msg3, 'other-attr': msg }"
      id="{ id: msg2, 'other-attr': msg }"
      .id="{ id: msg, 'other-attr': msg }"
      v-bind:src="msg"
      v-bind:[msg]="msg"
      :[msg]="msg"
      :xlink:special="msg">
          div {
            color: red;
          }
        </style>`,
    )
    expect(content).toMatch(`_useCEStyleAttrs(_ctx => ([
  {
   "id": { id: _ctx.msg3, 'other-attr': _ctx.msg },
   "src": _ctx.msg,
   [_ctx.msg]: _ctx.msg,
   "xlink:special": _ctx.msg,
 },
]))}`)
    expect(content).toMatch(
      `import { useCEStyleAttrs as _useCEStyleAttrs } from 'vue'`,
    )
    assertCode(content)
  })

  test('w/ <script setup> binding analysis', () => {
    const { content } = compileSFCScript(
      `<script setup>
        import { defineProps, ref } from 'vue'
        const msg = 'red'
        const msg2 = ref('10px')
        defineProps({
          msg3: String
        })
        </script>\n` +
        `<style :id="{ id: msg3, 'other-attr': msg }"
      id="{ id: msg2, 'other-attr': msg }"
      .id="{ id: msg, 'other-attr': msg }"
      v-bind:src="msg2"
      v-bind:[msg]="msg"
      :[msg]="msg3"
      :xlink:special="msg3">
          div {
            color: red;
          }
        </style>`,
    )
    // should handle:
    // 1. local const bindings
    // 2. local potential ref bindings
    // 3. props bindings (analyzed)
    expect(content).toMatch(`_useCEStyleAttrs(_ctx => ([
  {
   "id": { id: __props.msg3, 'other-attr': msg },
   "src": msg2.value,
   [msg]: msg,
   "xlink:special": __props.msg3,
 },
]))`)
    expect(content).toMatch(
      `import { useCEStyleAttrs as _useCEStyleAttrs, unref as _unref } from 'vue'`,
    )
    assertCode(content)
  })

  describe('codegen', () => {
    test('<script> w/ no default export', () => {
      assertCode(
        compileSFCScript(
          `<script>const msg = 1</script>\n` +
            `<style :id="{ id: msg3, 'other-attr': msg }"
id="{ id: msg2, 'other-attr': msg }"
.id="{ id: msg, 'other-attr': msg }"
v-bind:src="msg"
v-bind:[msg]="msg"
:[msg]="msg"
:xlink:special="msg">div{ color: red; }</style>`,
        ).content,
      )
    })

    test('<script> w/ default export', () => {
      assertCode(
        compileSFCScript(
          `<script>export default { setup() {} }</script>\n` +
            `<style :id="{ id: msg3, 'other-attr': msg }"
id="{ id: msg2, 'other-attr': msg }"
.id="{ id: msg, 'other-attr': msg }"
v-bind:src="msg"
v-bind:[msg]="msg"
:[msg]="msg"
:xlink:special="msg>div{ color: red }</style>`,
        ).content,
      )
    })

    test('<script> w/ default export in strings/comments', () => {
      assertCode(
        compileSFCScript(
          `<script>
          // export default {}
          export default {}
        </script>\n` +
            `<style :id="{ id: msg3, 'other-attr': msg }"
id="{ id: msg2, 'other-attr': msg }"
.id="{ id: msg, 'other-attr': msg }"
v-bind:src="msg"
v-bind:[msg]="msg"
:[msg]="msg"
:xlink:special="msg>div{ color: red }</style>`,
        ).content,
      )
    })

    test('w/ <script setup>', () => {
      assertCode(
        compileSFCScript(
          `<script setup>const msg = 'red'</script>\n` +
            `<style :id="{ id: msg3, 'other-attr': msg }"
id="{ id: msg2, 'other-attr': msg }"
.id="{ id: msg, 'other-attr': msg }"
v-bind:src="msg"
v-bind:[msg]="msg"
:[msg]="msg"
:xlink:special="msg>div{ color: red }</style>`,
        ).content,
      )
    })

    test('w/ <script setup> using the same var multiple times', () => {
      const { content } = compileSFCScript(
        `<script setup>
        const msg = 'red'
        </script>\n` +
          `<style :id="{ id: msg3, 'other-attr': msg }"
id="{ id: msg2, 'other-attr': msg }"
.id="{ id: msg, 'other-attr': msg }">
        
          p {
            color: red;
          }
        </style>`,
      )

      // id should only be injected once, even if it is twice in style
      expect(content).toMatch(`_useCEStyleAttrs(_ctx => ([
  {
   "id": { id: _ctx.msg3, 'other-attr': msg },
 },
]))`)
      assertCode(content)
    })

    test('should be able to parse incomplete expressions', () => {
      const {
        descriptor: { ceStyleAttrs },
      } = parse(
        `<script setup>let xxx = 1</script>
        <style scoped lang :id="xxx" :data-name="count.toString(">
        label {
         color:red
        }
        </style>`,
      )
      expect(ceStyleAttrs[0].length).toBe(2)
      expect(ceStyleAttrs[0]).toMatchObject([
        {
          name: 'bind',
          exp: {
            content: 'xxx',
            isStatic: false,
            constType: 0,
          },
          arg: {
            content: 'id',
            isStatic: true,
            constType: 3,
          },
        },
        {
          name: 'bind',
          exp: {
            content: 'count.toString(',
            isStatic: false,
            constType: 0,
          },
          arg: {
            content: 'data-name',
            isStatic: true,
            constType: 3,
          },
        },
      ])
    })

    test('It should correctly parse the case where there is no space after the script tag', () => {
      const { content } = compileSFCScript(
        `<script setup>import { ref as _ref } from 'vue';
                let background = _ref('red')
             </script>
             <style :id="background">
             h1 {color: red}
             </style>`,
      )
      expect(content).toMatch(
        `import { useCEStyleAttrs as _useCEStyleAttrs, unref as _unref } from 'vue'\nimport { ref as _ref } from 'vue';\n                \nexport default {\n  setup(__props, { expose: __expose }) {\n  __expose();\n\n_useCEStyleAttrs(_ctx => ([\n  {\n   \"id\": _unref(background),\n },\n]))\nlet background = _ref('red')\n             \nreturn { get background() { return background }, set background(v) { background = v }, _ref }\n}\n\n}`,
      )
    })

    test('With cssvars', () => {
      const { content } = compileSFCScript(
        `<script setup>import { ref as _ref } from 'vue';
                let background = _ref('red')
             </script>
             <style :id="background">
             label {
               background: v-bind(background);
             }
             </style>`,
      )
      expect(content).toMatch(
        `import { useCssVars as _useCssVars, unref as _unref, useCEStyleAttrs as _useCEStyleAttrs } from 'vue'\nimport { ref as _ref } from 'vue';\n                \nexport default {\n  setup(__props, { expose: __expose }) {\n  __expose();\n\n_useCEStyleAttrs(_ctx => ([\n  {\n   \"id\": _unref(background),\n },\n]))\n\n_useCssVars(_ctx => ({\n  \"xxxxxxxx-background\": (_unref(background))\n}))\nlet background = _ref('red')\n             \nreturn { get background() { return background }, set background(v) { background = v }, _ref }\n}\n\n}`,
      )
    })

    test('Multiple style tags', () => {
      const { content } = compileSFCScript(
        `<script setup>import { ref as _ref } from 'vue';
                let background = _ref('red')
             </script>
             <style :id="background">
             label {
               background: v-bind(background);
             }
             </style>
             <style :id="background">
             h1 {
               background: v-bind(background);
             }
             </style>`,
      )
      expect(content).toMatch(
        `import { useCssVars as _useCssVars, unref as _unref, useCEStyleAttrs as _useCEStyleAttrs } from 'vue'\nimport { ref as _ref } from 'vue';\n                \nexport default {\n  setup(__props, { expose: __expose }) {\n  __expose();\n\n_useCEStyleAttrs(_ctx => ([\n  {\n   \"id\": _unref(background),\n },\n  {\n   \"id\": _unref(background),\n },\n]))\n\n_useCssVars(_ctx => ({\n  \"xxxxxxxx-background\": (_unref(background))\n}))\nlet background = _ref('red')\n             \nreturn { get background() { return background }, set background(v) { background = v }, _ref }\n}\n\n}`,
      )
    })

    describe('skip codegen in SSR', () => {
      test('script setup, inline', () => {
        const { content } = compileSFCScript(
          `<script setup>
          let size = 1
          </script>\n` +
            `<style :id="size">
              div {
                color: red
              }
            </style>`,
          {
            inlineTemplate: true,
            templateOptions: {
              ssr: true,
            },
          },
        )
        expect(content).not.toMatch(`_useCssVars`)
      })

      // #6926
      test('script, non-inline', () => {
        const { content } = compileSFCScript(
          `<script setup>
          let size = 1
          </script>\n` +
            `<style :id="size">
              div {
                color: red
              }
            </style>\``,
          {
            inlineTemplate: false,
            templateOptions: {
              ssr: true,
            },
          },
        )
        expect(content).not.toMatch(`_useCssVars`)
      })

      test('normal script', () => {
        const { content } = compileSFCScript(
          `<script>
          export default {
            setup() {
              return {
                size: ref('100px')
              }
            }
          }
          </script>\n` +
            `<style :id="size">
              div {
                color: red
              }
            </style>\``,
          {
            templateOptions: {
              ssr: true,
            },
          },
        )
        expect(content).not.toMatch(`_useCssVars`)
      })
    })
  })
})
