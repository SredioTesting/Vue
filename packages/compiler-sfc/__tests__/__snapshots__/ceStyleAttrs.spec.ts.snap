// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CE style attrs injection > codegen > <script> w/ default export 1`] = `"export default { setup() {} }"`;

exports[`CE style attrs injection > codegen > <script> w/ default export in strings/comments 1`] = `
"
          // export default {}
          export default {}
        "
`;

exports[`CE style attrs injection > codegen > <script> w/ no default export 1`] = `
"const msg = 1
const __default__ = {}
import { useCEStyleAttrs as _useCEStyleAttrs } from 'vue'
const __injectCSSVars__ = () => {

_useCEStyleAttrs(_ctx => ([
  {
   "id": { id: _ctx.msg3, 'other-attr': _ctx.msg },
   "src": _ctx.msg,
   [_ctx.msg]: _ctx.msg,
   "xlink:special": _ctx.msg,
 },
]))}
const __setup__ = __default__.setup
__default__.setup = __setup__
  ? (props, ctx) => { __injectCSSVars__();return __setup__(props, ctx) }
  : __injectCSSVars__

export default __default__"
`;

exports[`CE style attrs injection > codegen > w/ <script setup> 1`] = `
"const msg = 'red'
export default {
  setup(__props, { expose: __expose }) {
  __expose();

return { msg }
}

}"
`;

exports[`CE style attrs injection > codegen > w/ <script setup> using the same var multiple times 1`] = `
"import { useCEStyleAttrs as _useCEStyleAttrs, unref as _unref } from 'vue'
const msg = 'red'
        
export default {
  setup(__props, { expose: __expose }) {
  __expose();

_useCEStyleAttrs(_ctx => ([
  {
   "id": { id: _ctx.msg3, 'other-attr': msg },
 },
]))

        
return { msg }
}

}"
`;

exports[`CE style attrs injection > generating correct code for nested paths 1`] = `
"const a = 1
const __default__ = {}
import { useCEStyleAttrs as _useCEStyleAttrs } from 'vue'
const __injectCSSVars__ = () => {

_useCEStyleAttrs(_ctx => ([
  {
   "id": { id: _ctx.msg3, 'other-attr': _ctx.msg },
   "src": _ctx.msg,
   [_ctx.msg]: _ctx.msg,
   "xlink:special": _ctx.msg,
 },
]))}
const __setup__ = __default__.setup
__default__.setup = __setup__
  ? (props, ctx) => { __injectCSSVars__();return __setup__(props, ctx) }
  : __injectCSSVars__

export default __default__"
`;

exports[`CE style attrs injection > w/ <script setup> binding analysis 1`] = `
"import { useCEStyleAttrs as _useCEStyleAttrs, unref as _unref } from 'vue'
import { ref } from 'vue'
        const msg = 'red'
        
export default {
  props: {
          msg3: String
        },
  setup(__props, { expose: __expose }) {
  __expose();

_useCEStyleAttrs(_ctx => ([
  {
   "id": { id: __props.msg3, 'other-attr': msg },
   "src": msg2.value,
   [msg]: msg,
   "xlink:special": __props.msg3,
 },
]))

        const msg2 = ref('10px')
        
        
return { msg, msg2, ref }
}

}"
`;

exports[`CE style attrs injection > w/ normal <script> binding analysis 1`] = `
"
      const __default__ = {
        setup() {
          return {
            msg: ref('100px')
          }
        }
      }
      
import { useCEStyleAttrs as _useCEStyleAttrs } from 'vue'
const __injectCSSVars__ = () => {

_useCEStyleAttrs(_ctx => ([
  {
   "id": { id: _ctx.msg3, 'other-attr': _ctx.msg },
   "src": _ctx.msg,
   [_ctx.msg]: _ctx.msg,
   "xlink:special": _ctx.msg,
 },
]))}
const __setup__ = __default__.setup
__default__.setup = __setup__
  ? (props, ctx) => { __injectCSSVars__();return __setup__(props, ctx) }
  : __injectCSSVars__

export default __default__"
`;
